app/user_agent.go

```diff
@@ -53,57 +54,64 @@ func getOSName(ua *uasurfer.UserAgent) string {
 		major := os.Version.Major
 		minor := os.Version.Minor
 
-		name := "Windows"
-
-		// Adapted from https://github.com/mssola/user_agent/blob/master/operating_systems.go#L26
-		if major == 5 {
-			if minor == 0 {
-				name = "Windows 2000"
-			} else if minor == 1 {
-				name = "Windows XP"
-			} else if minor == 2 {
-				name = "Windows XP x64 Edition"
-			}
-		} else if major == 6 {
-			if minor == 0 {
-				name = "Windows Vista"
-			} else if minor == 1 {
-				name = "Windows 7"
-			} else if minor == 2 {
-				name = "Windows 8"
-			} else if minor == 3 {
-				name = "Windows 8.1"
-			}
-		} else if major == 10 {
-			name = "Windows 10"
+		switch {
+		case major == 5 && minor == 0:
+			return "Windows 2000"
+		case major == 5 && minor == 1:
+			return "Windows XP"
+		case major == 5 && minor == 2:
+			return "Windows XP x64 Edition"
+		case major == 6 && minor == 0:
+			return "Windows Vista"
+		case major == 6 && minor == 1:
+			return "Windows 7"
+		case major == 6 && minor == 2:
+			return "Windows 8"
+		case major == 6 && minor == 3:
+			return "Windows 8.1"
+		case major == 10:
+			return "Windows 10"
+		default:
+			return "Windows"
 		}
+	}
 
+	name, ok := osNames[os.Name]
+	if ok {
 		return name
-	} else if name, ok := osNames[os.Name]; ok {
-		return name
-	} else {
-		return osNames[uasurfer.OSUnknown]
 	}
+
+	return osNames[uasurfer.OSUnknown]
 }
 
 func getBrowserVersion(ua *uasurfer.UserAgent, userAgentString string) string {
```

Comment:
Here you are changing more things that I would expect from this PR. The first set of ifs with a switch is pretty clear, but in this case is not so clear, so I would simply use the previous code using the if + return, not the new function.

Reply:
The function was using the same code in the if-clause so I extracted it into its own function, but sure I will remove the new function and simply use if without the else

Reply:
done